src/@types/global.d.ts

```diff
@@ -1,27 +1,19 @@
-export type Viewport = {
-  id: string,
-  containerElement: Element,
-  svgElement: Element,
+export type Context = {
+  canvas: {
+    screen: HTMLCanvasElement,
+    offscreen: HTMLCanvasElement,
```

Comment:
what is the role of offscreen?

Reply:
offscreen canvas is **rendering performance improvement methodology**.
and i also used to cope with various resolution.
in the future, it is planned to connected with web worker.

**link for detail**
https://developers.google.com/web/updates/2018/08/offscreen-canvas